api/internal/core/server/server.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/apisix/manager-api/internal/conf"
+	"github.com/apisix/manager-api/internal/log"
+	"github.com/apisix/manager-api/internal/utils"
+)
+
+// server is the manager of Manager API, which is responsible for managing the life cycle of Manager API, including initialization, start, stop and so on
+type server struct {
+	server    *http.Server
+	serverSSL *http.Server
+	options   *Options
+}
+
+type Options struct {
+	ForceStart bool // force start new instance
+}
+
+// NewServer Create a server manager
+func NewServer(options *Options) (*server, error) {
+	return &server{options: options}, nil
+}
+
+func (s *server) Start() error {
+	// initialize server
+	err := s.init()
+	if err != nil {
+		return err
+	}
+
+	// write daemon pid file
+	err = s.writePID()
+	if err != nil {
+		return err
+	}
+
+	// print server info to stdout
+	s.printInfo()
+
+	// For internal error handling across multiple goroutines.
+	errSig := make(chan error, 1)
+
+	// start HTTP server
+	log.Infof("The Manager API is listening on %s", s.server.Addr)
+	go func() {
+		if err := s.server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Errorf("listen and serv fail: %s", err)
+			errSig <- err
+		}
+	}()
+
+	// start HTTPs server
+	if conf.SSLCert != "" && conf.SSLKey != "" {
+		go func() {
+			err := s.serverSSL.ListenAndServeTLS(conf.SSLCert, conf.SSLKey)
+			if err != nil && err != http.ErrServerClosed {
+				log.Errorf("listen and serve for HTTPS failed: %s", err)
+				errSig <- err
+			}
+		}()
+	}
+
+	// handle HTTP(s) server error
+	select {
+	case err := <-errSig:
+		return err
+	default:
+		return nil
+	}
+}
+
+func (s *server) Stop() {
+	utils.CloseAll()
+
+	s.shutdownServer(s.server)
+	s.shutdownServer(s.serverSSL)
+}
+
+func (s *server) init() error {
+	log.Info("Initialize Manager API store")
+	err := s.setupStore()
+	if err != nil {
+		return err
+	}
+
+	log.Info("Initialize Manager API server")
+	s.setupHTTPServer()
```

Comment:
Only setup the HTTP server is enough?

Reply:
In fact, it initializes routes, HTTP(s) server and any other content related to HTTP API. I don't have a better idea of naming it.
Do you have any suggestions?

Reply:
Then the name is confusing, what about just `s.setupServers`?

Reply:
emmm, If it's `s.setupServers`, it looks like it's initializing the entire `Manager API server`.

Reply:
Or set up API.

Reply:
I think it's good